test-specifications/cryptoService/accountCreateTransaction.md

```diff
@@ -0,0 +1,155 @@
+
+**Response codes:**
+
+https://github.com/hashgraph/hedera-protobufs/blob/main/services/response_code.proto
+
+**Mirror Node APIs:**
+
+https://docs.hedera.com/hedera/sdks-and-apis/rest-api
+
+## Initialisation
+
+```jsx
+new AccountCreateTransaction()
+
+//example
+
+const transaction = new AccountCreateTransaction()
+  .setKey(privateKey.publicKey)
+  .setInitialBalance(new Hbar(1000))
+  ...
+```
+
+## JSON-RPC API Endpoint Documentation
+
+### Method Name
+
+`createAccount`
+
+### Parameters
+
+| Parameter Name            | Type   | Required/Optional | Description/Notes                                                                                                                           |
+|---------------------------|--------|-------------------|---------------------------------------------------------------------------------------------------------------------------------------------|
+| key                       | string | optional          | DER-encoded hex string representation for private or public keys. Keylists and threshold keys are the hex of the serialized protobuf bytes. |
+| initialBalance            | int64  | optional          | Units of tinybars                                                                                                                           |
+| receiverSignatureRequired | bool   | optional          |                                                                                                                                             |
+| autoRenewPeriod           | int64  | optional          | Units of seconds                                                                                                                            |
+| memo                      | string | optional          |                                                                                                                                             |
+| maxAutoTokenAssociations  | int32  | optional          |                                                                                                                                             |
+| stakedAccountId           | string | optional          |                                                                                                                                             |
+| stakedNodeId              | int64  | optional          |                                                                                                                                             |
+| declineStakingReward      | bool   | optional          |                                                                                                                                             |
+| alias                     | string | optional          | Hex string representation of a serialized protobuf Key of an ED25519 or ECDSAsecp256k1 public key type.                                     |
```

Comment:
If you are setting the alias on account create you can only set an ECDSAsecp256k1 public key type.

From the HIP:

```
message CryptoCreateTransactionBody {
	...
	/**
     * The bytes to be used as the account's alias. 
     *
     * The bytes must be formatted as the calcluated last 20 bytes of the
     * keccak-256 of the ECDSA primitive key.</li>
     *
     * All other types of keys, including but not limited to ED25519, ThresholdKey, KeyList, ContractID, and
     * delegatable_contract_id, are not supported.
     *
     * At most only one account can ever have a given alias on the network.
     */
	bytes alias = 18;
}

```

> All other types of keys, including but not limited to ED25519, ThresholdKey, KeyList, ContractID, and
>      * delegatable_contract_id, are not supported.

Reply:
https://hips.hedera.com/hip/hip-583

Reply:
Ah forgot to update this, good catch! I'll update it now

Reply:
Fixed